support-frontend/assets/helpers/patrons.ts

```diff
@@ -0,0 +1,34 @@
+import type {
+	CountryGroupId,
+	CountryGroupName,
+} from './internationalisation/countryGroup';
+import { countryGroups } from './internationalisation/countryGroup';
+import type { IsoCurrency } from './internationalisation/currency';
+import { NoFulfilmentOptions } from './productPrice/fulfilmentOptions';
+import { NoProductOptions } from './productPrice/productOptions';
+import type { ProductPrices } from './productPrice/productPrices';
+import type { Promotion } from './productPrice/promotions';
+
+export function getPromotions(
+	countryGroupId: CountryGroupId,
+	productPrices?: ProductPrices,
+	currencyId?: IsoCurrency,
+): Promotion[] | undefined {
+	if (!productPrices || !currencyId) return;
+
+	const countryGroupName: CountryGroupName = countryGroups[countryGroupId].name;
+	const billingPeriods =
+		productPrices[countryGroupName][NoFulfilmentOptions][NoProductOptions];
+
+	return Object.values(billingPeriods)[0][currencyId].promotions;
```

Comment:
I'm slightly unsure how this line works. Is it grabbing either the Annual or Monthly value?

Reply:
Yeah here we access the promotions for either the annual or monthly plan dependant on which is the first element in the array. We then pass the output of this function into the `userIsPatron` function where we're only concerned if the `numberOfDiscountedPeriods` is 100 or more, which will be the case for any plan if a user is a patron

Reply:
Ah okay that makes sense - given the use of this function we don't actually care if we grab the monthly or annual promotions, so it's fine to get either. That being said, I think it might be worth refactoring a little here as that's not captured in the name of the function. I could imagine a few months from now trying to reuse this in a scenario where it is important if we use annual/monthly. Maybe this function should also take an argument that is the frequency?

Reply:
Have updated this function to take the billing period as an arg

Reply:
Nice - yeah I think that makes it align better with the name and makes it more reusable later on!